torchaudio/_internal/module_utils.py

```diff
@@ -43,7 +43,11 @@ def deprecated(direction: str, version: Optional[str] = None):
     Args:
         direction: Migration steps to be given to users.
     """
-    def decorator(func):
+    def decorator(obj):
+        # get __init__ if obj is a class, else get the function itself
+        # FIXME: remove when SignalInfo and EncodingInfo are properly removed
+        func = getattr(obj, '__init__', obj)
```

Comment:
For ref, this is needed to avoid a similar error as in https://github.com/astropy/astropy/issues/2051.

The main issue is that `decorator` doesn't  work on classes; it only works on functions but it's currently used for the `SignalInfo` and `EncodingInfo` classes which led to the warning / error.

I'm not sure the current changes properly makes the decorator work on classes - but it does successfully suppress the warning, and as the offending classes are deprecated and will be removed, I assumed it was good enough.

If needed, I'm happy to propose a proper fix and let the decorator work on classes, but that will be a bit more work

Reply:
Thanks for the clarification. `SignalInfo` and `EncodingInfo` classes are expected to be removed after the next release (if things go smoothly) so I think this workaround is good enough.

Reply:
I couldn't make it work properly after all, so I removed the changes to the decorator and removed its us in the corresponding classes, and instead warned inside `__init__`

Reply:
Sounds good.